sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/reinterpret/BeamSqlReinterpretExpression.java

```diff
@@ -47,6 +48,13 @@ public BeamSqlReinterpretExpression(List<BeamSqlExpression> operands, SqlTypeNam
 
   @Override
   public boolean accept() {
+    // Interval types will be already converted into BIGINT after evaluation.
+    SqlTypeFamily opTypeFamily = opType(0).getFamily();
+    if (opTypeFamily.equals(SqlTypeFamily.INTERVAL_DAY_TIME)
```

Comment:
I think it's good to add a unit test for it

Reply:
@vectorijk Actually, it's already implicitly tested by `BeamSqlDslSqlStdOperatorsTest.testTimestampDiff()` and it failed after I did a refactoring of `BeamSqlFnExecutor` and before adding this check. Do you think it will make sense to add specific test for this case?